### DISCUSSION TITLE
Store duplicated rows in separate table

### DISCUSSION THREAD

**Discussion by treasurely** [supabase/supabase#9623] [Questions]
Does anyone have advice on handling the following scenario -- a simplified version of our set up is:

We have two tables that track payment imported into the app -- "New Payments" & "Duplicate Payments".  We are looking to insert/upsert payments imported from a csv into a table, and if there are duplicates present in the table the data is not inserted but is instead added to "Duplicate Payments" so a user could review & audit before letting the payment through?  The duplicate check is done on the account & internal reference columns:

e.g. a list of payments called in the insert/upsert function:

```
[{date: '2020-01-01', account: 'abc123', amount: 100, internalRefernce: '9283asjlc358'},
 {date: '2020-01-02', account: 'bcd456', amount: 200, internalRefernce: 'ajsc28385'},
 {date: '2020-01-05', account: 'def789', amount: 300, internalRefernce: 'elvaiopsdj923'}]

```

If there is already an existing payment for account 'def789' with internal reference 'elvaiopsdj923', is there a correct way to run an upsert/insert function to insert the first two rows, then insert the third row into "Duplicate Payments"?

**Comment by Sukikiroi** [comment]
Hello I try this ,Maybe gona Help you good luck.

fs.createReadStream("./example.csv")
  .pipe(parse({ delimiter: ",", from_line: 2 }))
  .on("data", function (row) {
//here we are exploring csv file by row
const { data, error } = await supabase
  .from('New Payments')
  .select()
  .filter('account', 'eq', row.account)
  .filter('internalRefernce', 'eq', row.internalRefernce)
if(!data){
// Insert your csv row into  New Payments Table
}
else{
//Insert your csv row into  Duplicate Payments Table 
}

  })
  .on("error", function (error) {
    console.log(error.message);
  })
  .on("end", function () {
    console.log("finished");
  });
